src/python/doc/perslay.md

```diff
@@ -0,0 +1,135 @@
+# How to call and use PersLay
+
+
+The python function `perslay` takes four arguments:
+
+  * the first three ones are: **output** which is the list that will contain the output of PersLay, **name** which is a string defining the name of the PersLay operation for tensorflow, and **diag** which is a numpy array containing the persistence diagrams.
+
+  * the fourth argument is a python dictionary containing the parameters of PersLay. Examples can be found in `tutorialPersLay.ipynb`.
+ 
+In the following description of PersLay parameters, each parameter, or dictionary key, that contains "_init" in its name is optimized and learned by PersLay during training. If you do not want to optimize the vectorization, set the (optional) keys **train_vect** and **train_weight** to False.
+
+  * The following keys are mandatory:
+
+    | **name** | **description** |
+    | --- | --- |
+    | **layer**              | Either "pm", "im", "ls", "bc", "en", "ex", "rt", "rh". Type of the PersLay layer. "im" is for [persistence images](https://arxiv.org/abs/1507.06217), "ls" is for [persistence landscapes](http://www.jmlr.org/papers/volume16/bubenik15a/bubenik15a.pdf), "ex", "rt" and "rh" are for [structure elements](http://jmlr.org/beta/papers/v20/18-358.html), "pm" is for the original DeepSet layer, defined in [this article](https://arxiv.org/abs/1703.06114), "bc" is for [Betti curves](https://www.jstage.jst.go.jp/article/tjsai/32/3/32_D-G72/_pdf) and "en" is for [entropy](https://arxiv.org/abs/1803.08304). |
+    | **perm_op**            | Either "sum", "mean", "max", "topk". Permutation invariant operation. |
```

Comment:
Isn't "reduction" a standard name for this kind of operation?

Reply:
Well, "sort" could be an acceptable `perm_op` I think. `top-k` is just a "take the k largest entries" so it already looks like a sort (if I remember correctly).